route_guide/route_guide_server.cc

```diff
@@ -55,133 +64,235 @@ float GetDistance(const Point& start, const Point& end) {
   float lon_2 = end.longitude() / kCoordFactor;
   float lat_rad_1 = ConvertToRadians(lat_1);
   float lat_rad_2 = ConvertToRadians(lat_2);
-  float delta_lat_rad = ConvertToRadians(lat_2-lat_1);
-  float delta_lon_rad = ConvertToRadians(lon_2-lon_1);
+  float delta_lat_rad = ConvertToRadians(lat_2 - lat_1);
+  float delta_lon_rad = ConvertToRadians(lon_2 - lon_1);
 
-  float a = pow(sin(delta_lat_rad/2), 2) + cos(lat_rad_1) * cos(lat_rad_2) *
-            pow(sin(delta_lon_rad/2), 2);
-  float c = 2 * atan2(sqrt(a), sqrt(1-a));
+  float a = pow(sin(delta_lat_rad / 2), 2)
+      + cos(lat_rad_1) * cos(lat_rad_2) * pow(sin(delta_lon_rad / 2), 2);
+  float c = 2 * atan2(sqrt(a), sqrt(1 - a));
   int R = 6371000; // metres
 
   return R * c;
 }
 
-std::string GetFeatureName(const Point& point,
-                           const std::vector<Feature>& feature_list) {
+// Just return the name of feature in feature_list vector
+std::string GetFeatureName(
+    const Point& point,
+    const std::vector<Feature>& feature_list) {
   for (const Feature& f : feature_list) {
-    if (f.location().latitude() == point.latitude() &&
-        f.location().longitude() == point.longitude()) {
+    if (
+        f.location().latitude() == point.latitude()
+        && f.location().longitude() == point.longitude()) {
       return f.name();
     }
   }
   return "";
 }
 
+
+// The idea of the RouteGuideImpl class is based on the stout-grpc interface
+// which uses callbacks. stout-grpc is intented to be a higher-level interface
+// while still being asynchronous that uses stout-eventuals
+// (https://github.com/3rdparty/stout-eventuals) to make composing async code
+// easier.
+// This class contains 4 methods: ServeGetFeature(unary rpc), ServeListFeatures
+// (server streaming), ServeRecordRoute(client streaming), ServeRouteChat
+// (bidirectional streaming).
```

Comment:
Good comment, just some English suggestions:
```suggestion
// The RouteGuideImpl class demonstrates server-side use of stout-grpc,
// which is a higher-level interface on top of gRPC while still being asynchronous.
// It uses stout-eventuals (https://github.com/3rdparty/stout-eventuals) to make 
// composing async code easier.
//
// This class contains 4 methods: ServeGetFeature (unary rpc), ServeListFeatures
// (server streaming), ServeRecordRoute (client streaming), and ServeRouteChat
// (bidirectional streaming).
```